pkgs/tools/misc/grub/pvgrub_image/default.nix

```diff
@@ -0,0 +1,42 @@
+{ stdenv, grub2_xen }:
+
+with stdenv.lib;
+let
+  efiSystemsBuild = {
+    "i686-linux".target = "i386";
+    "x86_64-linux".target = "x86_64";
+    "aarch64-linux".target = "aarch64";
+  };
+
+in (
+
+stdenv.mkDerivation rec {
+  name = "pvgrub-image";
+
+  configs = ./configs;
+
+  buildInputs = [ grub2_xen ];
+
+  buildCommand = ''
+    cp "${configs}"/* .
+    tar -cf memdisk.tar grub.cfg
+    # We include all modules except all_video.mod as otherwise grub will fail printing "no symbol table"
```

Comment:
The comment is based on my observation that pvgrub fails when we include that module. The website that you linked to concerns booting Xen itself using EFI, which requires compiling grub as an EFI loader. The `all_video.mod` module is probably working in that setup, which does not contradict what I wrote. Please let me know if I should include any clarification in the comment.